lib/src/list.dart

```diff
@@ -85,15 +85,16 @@ class ManagedRealmList<T extends Object?> with RealmEntity, ListMixin<T> impleme
 
   @override
   bool remove(covariant T element) {
-    if (element is RealmObject && !element.isManaged) {
+    if (element is RealmObjectBase && !element.isManaged) {
       throw RealmStateError('Cannot call remove on a managed list with an element that is an unmanaged object');
     }
     final index = indexOf(element);
-    final found = index > 0;
-    if (found) {
-      removeAt(index);
+    if (index < 0) {
+      return false;
     }
-    return found;
+
+    removeAt(index);
+    return true;
```

Comment:
This is a fix for `list.remove` not removing the first element (due to the check being `found = index > 0`). It came up when testing embedded lists, so decided to fix it as part of this PR rather than file a separate one just for this.